opcua.cpp

```diff
@@ -250,3 +309,71 @@ void OPCUAServer::stop()
 		m_server->Stop();
 	}
 }
+
+OpcUa::Node& OPCUAServer::findParent(const Reading *reading)
+{
+	vector<Datapoint *> datapoints = reading->getReadingData();
+	for (int i = 0; i < datapoints.size(); i++)
+	{
+		string name = datapoints[i]->getName();
+		for (int j = 0; j < m_hierarchy.size(); j++)
+		{
+			if (m_hierarchy[j].getName().compare(name) == 0)
+			{
+				DatapointValue value = datapoints[i]->getData();
+				string svalue = value.toString();
+				svalue = svalue.substr(1, svalue.length() - 2);
+				OpcUa::Node opcNode;
+				auto it = m_parents.find(svalue);
+				if (it != m_parents.end())
+				{
+					opcNode = it->second;
+				}
+				else
+				{
+					NodeId  nodeId(svalue, m_idx);
+					QualifiedName   qn(svalue, m_idx);
+					opcNode = m_objects.AddObject(nodeId, qn);
+					m_parents.insert(pair<string, Node>(svalue, opcNode));
+				}
+				return findParent(m_hierarchy[j].getChildren(), reading, opcNode, svalue);
+			}
+		}
+	}
+	return m_objects;
+}
+
```

Comment:
Documentation header would help the reader

Reply:
That's odd, in my code there is a comment